src/AppInstallerCLICore/Commands/SearchCommand.cpp

```diff
@@ -64,6 +65,14 @@ namespace AppInstaller::CLI
         return "https://aka.ms/winget-command-search";
     }
 
+    void SearchCommand::ValidateArgumentsInternal(Args& execArgs) const
+    {
+        if (execArgs.Contains(Args::Type::CustomHeader) && !execArgs.Contains(Args::Type::Source))

```

Comment:
>if (execArgs.Contains(Args::Type::CustomHeader) && !execArgs.Contains(Args::Type::Source)) [](http://example.com/codeflow?start=8&length=90)

Since this logic is common and copied into multiple commands, we could probably move them up in base Command::ValidateArguments as where we do interactive|silent check